test/infrastructure/docker/api/v1beta1/dockermachinetemplate_webhook_test.go

```diff
@@ -51,10 +60,22 @@ func TestDockerMachineTemplateInvalid(t *testing.T) {
 			oldTemplate: &oldTemplate,
 			wantError:   true,
 		},
+		{
+			name:        "skip immutability check",
+			newTemplate: newTemplateSkipImmutability,
+			oldTemplate: &oldTemplate,
+			req:         &admission.Request{AdmissionRequest: admissionv1.AdmissionRequest{DryRun: pointer.Bool(true)}},
+			wantError:   false,
+		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			err := tt.newTemplate.ValidateUpdate(tt.oldTemplate)
+			wh := &DockerMachineTemplateWebhook{}
+			ctx := context.TODO()
```

Comment:
Let's please use context.Background() in unit tests.

context.TODO() is supposed to be a TODO in the sense of that eventually a real context should be propagated to the place where context.TODO() is used and then replace the TODO